ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/ForkIdManager.java

```diff
@@ -205,16 +230,20 @@ private static long createForkIds(
     final CRC32 crc = new CRC32();
     crc.update(genesisHash.toArray());
     final List<Bytes> forkHashes = new ArrayList<>(List.of(getCurrentCrcHash(crc)));
-    for (final Long fork : forks) {
-      updateCrc(crc, fork);
-      forkHashes.add(getCurrentCrcHash(crc));
-    }
+    final List<Long> filteredForks =
+        forks.stream().filter(v -> v > 0).distinct().collect(Collectors.toUnmodifiableList());
```

Comment:
Do we need this filter as well? I thought we already got rid of the 0's above. If it does break when this is removed, I wonder if there's a way we can do this filter in only one place.

Reply:
Fixed.